lib/device.c

```diff
@@ -566,15 +566,13 @@ int sync_update(struct sync_device *d, int row, struct sync_cb *cb,
 		}
 	}
 
-	if (cb && cb->is_playing && cb->is_playing(cb_param)) {
-		if (d->row != row && d->sock != INVALID_SOCKET) {
-			unsigned char cmd = SET_ROW;
-			uint32_t nrow = htonl(row);
-			if (xsend(d->sock, (char*)&cmd, 1, 0) ||
-			    xsend(d->sock, (char*)&nrow, sizeof(nrow), 0))
-				goto sockerr;
-			d->row = row;
-		}
+	if (d->row != row && d->sock != INVALID_SOCKET) {
+		unsigned char cmd = SET_ROW;
+		uint32_t nrow = htonl(row);
+		if (xsend(d->sock, (char*)&cmd, 1, 0) ||
+		    xsend(d->sock, (char*)&nrow, sizeof(nrow), 0))
+			goto sockerr;
+		d->row = row;
```

Comment:
I'm not sure I'm convinced that this is a good idea. Without this, I've seen glitches when pausing/unpausing due to races about taking over control. We didn't add this for no reason. But thinking about it a bit more, the way we currently do it isn't 100% race-proof either, but it's better than nothing.

This is also the only call-site of `struct sync_cb::is_playing`, so if this is a good idea after all, I think it should probably also remove it, or mark it as deprecated somehow.

Reply:
I don't know the history, but considering the only effect of !is_playing() is preventing SET_ROW from getting sent to the client when it changes, how does that prevent race conditions?  The client doesn't even receive the result of is_playing(), do some clients infer a playing vs. paused state based on SET_ROW arriving?

Can't we just require clients to always be ready to accept SET_ROW to report the demo's latest position, without it being any indication of the paused vs. playing state?

I left the callback entry as I didn't want to break any existing users, my changes just added some stuff.

I'll clean up the other stuff you pointed out and force push an update, thanks for the review!

Also, I just noticed my demo-side PAUSE request didn't include the flag, it probably should include that.